test/helpers/foundry/foundryup.ts

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env -S node --require "./node_modules/tsx/dist/preflight.cjs" --import "./node_modules/tsx/dist/loader.mjs"
+
+import { join, relative } from 'node:path';
+import { homedir } from 'node:os';
+import { Dir, readFileSync } from 'node:fs';
+import { copyFile, opendir, rm, symlink, unlink } from 'node:fs/promises';
+import { createHash } from 'node:crypto';
+import { cwd, exit } from 'node:process';
+import { parse as parseYaml } from 'yaml';
+import {
+  extractFrom,
+  getVersion,
+  isCodedError,
+  noop,
+  parseArgs,
+  printBanner,
+  say,
+  transformChecksums,
+} from './helpers';
+import { Architecture, Checksums, Extension, Platform } from './types';
+
+export function getCacheDirectory(): string {
+  let enableGlobalCache = false;
+  try {
+    const configFileContent = readFileSync('.yarnrc.yml', 'utf8');
+    const parsedConfig = parseYaml(configFileContent);
+    enableGlobalCache = parsedConfig?.enableGlobalCache ?? false;
+  } catch (error) {
+    console.error('Error reading/parsing .yarnrc.yml:', error);
+  }
+  return enableGlobalCache
+    ? join(homedir(), '.cache', 'metamask')
+    : join(cwd(), '.metamask', 'cache');
+}
+
+export function getBinaryArchiveUrl(
+  repo: string,
+  tag: string,
+  version: string,
+  platform: Platform,
+  arch: string,
+): string {
+  const ext = platform === Platform.Windows ? Extension.Zip : Extension.Tar;
+  return `https://github.com/${repo}/releases/download/${tag}/foundry_${version}_${platform}_${arch}.${ext}`;
+}
+
+export async function checkAndDownloadBinaries(
+  url: URL,
+  binaries: string[],
+  cachePath: string,
+  platform: Platform,
+  arch: Architecture,
+  checksums: Checksums = {
+    algorithm: 'sha256',
+    binaries: {
+      anvil: {
+        'win32-amd64': '',
+        'win32-arm64': '',
+        'linux-amd64': '',
+        'linux-arm64': '',
+        'darwin-amd64': '',
+        'darwin-arm64': '',
+      },
+      forge: {
+        'win32-amd64': '',
+        'win32-arm64': '',
+        'linux-amd64': '',
+        'linux-arm64': '',
+        'darwin-amd64': '',
+        'darwin-arm64': '',
+      },
+      cast: {
+        'win32-amd64': '',
+        'win32-arm64': '',
+        'linux-amd64': '',
+        'linux-arm64': '',
+        'darwin-amd64': '',
+        'darwin-arm64': '',
+      },
+      chisel: {
+        'win32-amd64': '',
+        'win32-arm64': '',
+        'linux-amd64': '',
+        'linux-arm64': '',
+        'darwin-amd64': '',
+        'darwin-arm64': '',
+      },
```

Comment:
Ah, your testing caught a bug, I think.

I intended to make the `checksums` CLI option optional. Do you want me to update the code?

Reply:
oh yeah sure :raised_hands: thank you!!